packages/client-core/src/user/services/AuthService.ts

```diff
@@ -133,116 +137,87 @@ const resolveWalletUser = (credentials: any): UserType => {
-        const data = JSON.parse(e.data)
-        if (data.skipCrossOriginCookieCheck != null || data.storageAccessPermission === 'denied')
-          localStorage.setItem('skipCrossOriginCookieCheck', 'true')
-        resolve(data)
-      }
-    }
-    window.addEventListener('message', hasAccessListener)
-  })) as HasAccessType
-
-  if (!hasAccess.cookieSet || !hasAccess.hasStorageAccess) {
+  const hasAccess = (await communicator
+    .sendMessage('checkAccess')
+    .then((message) => {
+      if (message?.data?.skipCrossOriginCookieCheck === true || message?.data?.storageAccessPermission === 'denied')
+        localStorage.setItem('skipCrossOriginCookieCheck', 'true')
+      return message.data
+    })
+    .catch((message) => {
+      invalidDomainHandling(message)
+      return {}
+    })) as HasAccessType
+
+  if (!hasAccess?.cookieSet || !hasAccess?.hasStorageAccess) {
     const skipCheck = localStorage.getItem('skipCrossOriginCookieCheck')
     const invalidCrossOriginDomain = localStorage.getItem('invalidCrossOriginDomain')
     if (skipCheck === 'true' || invalidCrossOriginDomain === 'true') {
       const authState = getMutableState(AuthState)
       const accessToken = authState?.authUser?.accessToken?.value
       return Promise.resolve(accessToken?.length > 0 ? accessToken : '')
     } else {
-      iframe.style.display = 'block'
+      iframe.style.visibility = 'visible'
       return await new Promise((resolve) => {
         const clickResponseListener = async function (e) {
-          if (isRootCookieAncestorMessage(e)) {
-            try {
-              window.removeEventListener('message', clickResponseListener)
-              const parsed = !e.data ? {} : JSON.parse(e.data)
-              if (parsed.skipCrossOriginCookieCheck != null) {
-                localStorage.setItem('skipCrossOriginCookieCheck', parsed.skipCrossOriginCookieCheck)
-                iframe.style.display = 'none'
-                resolve('')
-              } else {
-                const token = await waitForToken(win, clientUrl)
-                iframe.style.display = 'none'
-                resolve(token)
-              }
-            } catch (err) {
-              //Do nothing
+          if (e.origin !== config.client.clientUrl || e.source !== iframe.contentWindow) return
+          try {
+            window.removeEventListener('message', clickResponseListener)
+            const data = e?.data?.data
+            if (data.skipCrossOriginCookieCheck === true || data.storageAccessPermission === 'denied') {
+              localStorage.setItem('skipCrossOriginCookieCheck', 'true')
+              iframe.style.visibility = 'hidden'
+              resolve('')
+            } else {
+              const token = await waitForToken(win, clientUrl)
+              iframe.style.visibility = 'hidden'
+              resolve(token)
             }
+          } catch (err) {
+            //Do nothing
```

Comment:
This catch won't ever be executed if anything goes wrong inside waitForToken for the reasons I mentioned before.

If this logic is triggered by a click event listener (user interaction) we need this catch to be executed on exceptions and to provide the user with feedback.